src/core/audit-fd.c

```diff
@@ -37,6 +38,13 @@ static int audit_fd;
 int get_audit_fd(void) {
 
         if (!initialized) {
+                if (have_effective_cap(CAP_AUDIT_WRITE) == 0) {
+                        audit_fd = -1;
```

Comment:
can you store -EPERM here, rather than -1? THis function is supposed to return a valid error code on failure, and not -1... 

(yes, -EPERM happens to be identical to -1 on Linux, but that's just a coincidence, we really shouldn't use numeric error codes but stick to Exxx ones...)